source/tools/testConsistency/dotnetWPF-renderer/WPFRenderer/WPFRenderer/Program.cs

```diff
@@ -0,0 +1,112 @@
+using AdaptiveCards.Templating;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WPFRenderer
+{
+    public class Program
+    {
+        public static void Main()
+        {
+            JObject resultJson = new JObject();
+            List<string> errorList = new List<string>();
+            List<string> warningList = new List<string>();
+            resultJson.Add("imageData", "");
+            resultJson.Add("warnings", "");
+            resultJson.Add("errors", "");
+            Console.OutputEncoding = Encoding.UTF8;
+            try
+            {
+                StringBuilder inputSb = new StringBuilder();
+                string currLine;
+
+                //read a multi line input from stdin
+                do
+                {
+                    currLine = Console.ReadLine();
+                    inputSb = inputSb.Append(currLine);
+                }
+                while ((currLine != null) && (currLine != ""));
+
+                JObject jsonInput = JObject.Parse(inputSb.ToString());
+                string cardData = jsonInput.GetValue("data").ToString();
+                string cardTemplate = jsonInput.GetValue("template").ToString();
+
+                AdaptiveCardTemplate adaptiveCardTemplate = new AdaptiveCardTemplate(cardTemplate);
+                string adaptiveCardJson = adaptiveCardTemplate.Expand(cardData);
+                AdaptiveCardParseResult parseResult = AdaptiveCard.FromJson(adaptiveCardJson);
+                AdaptiveCardRenderer renderer = new AdaptiveCardRenderer();
+
+                GetResultJsonString(parseResult, renderer, resultJson, warningList).GetAwaiter().GetResult();
+            }
+
+            catch (Newtonsoft.Json.JsonReaderException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The input is not a valid json");
+            }
+            catch (AdaptiveTemplateException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to bind the data with the template. The supplied card data is not valid");
+            }
+            catch (AdaptiveSerializationException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The supplied json input is not a valid Adaptive Card");
+
+            }
+            catch (ArgumentNullException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to capture the image of the card. Ensure that the card template and card data are valid and try again.");
+            }
+            catch (Exception e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("An error was encountered: " + e.Message);
+            }
```

Comment:
These 2 cases actually look like these should result in failing the actual POST request eventually - is that the case? Do errors end up completing the originating client request with a non-200 (OK) response or does the client always have to dig into the POST response body to figure that out?

Reply:
The response code will be 200. The client will look into the body and display the error it received.